tests/integration/projects/rust_edge_cases_project/src/main.rs

```
@@ -0,0 +1,36 @@
+// Binary crate root. Exercises:
+//   - main.rs stem collapse (functions here should be `src.main.<name>`)
+//   - cross-module function calls (edges to models, services, utils)
+//   - trait method invocation through dyn dispatch
+
+mod models;
+mod services;
+mod utils;
+
+use models::base::{Entity, Speaker};
```

Clippy (1.94.1)

[warning] unused import

(warning)